### ISSUE TITLE
Progress bar jumps from 0% to 100% when working with 1K+ local Parquet files

### ISSUE THREAD

**Issue by marklit** [duckdb/duckdb#11071]
### What happens?

I have 1,200 Parquet files of roughly equal size that total 56 GB altogether. Pretty much every operation I can run on them will bring up the progress bar where it'll sit at 0% and jump to 100% once the operation is complete. Is there a way to get more granual indication of progress?

### To Reproduce

```sql
select count(*) from read_parquet('/run/media/deck/Seagate/gis/Global/adsb_lol/traces*.pq');

select distinct ownOp from read_parquet('/run/media/deck/Seagate/gis/Global/adsb_lol/traces*.pq') order by 1;

select ownOp, COUNT(*)
from read_parquet('/run/media/deck/Seagate/gis/Global/adsb_lol/traces*.pq') 
WHERE trace.h3_5 = '851f6073fffffff'
group by 1
order by 2 DESC;
```

### OS:

Arch Linux on Stream Deck

### DuckDB Version:

v0.10.0 20b1486d11

### DuckDB Client:

CLI

### Full Name:

Mark Litwintschik

### Affiliation:

Green Idea

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a release build (and could not test with a nightly build)

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Mytherin** [comment]
Thanks for the report! I've pushed a fix in #11072

**Comment by marklit** [comment]
Great, thanks for the quick turn around.
